include/sharg/parser.hpp

```diff
@@ -903,6 +878,42 @@ class parser
         if (detail::format_parse::is_empty_id(short_id) && detail::format_parse::is_empty_id(long_id))
             throw design_error("Option Identifiers cannot both be empty.");
     }
+
+    //!brief Verify if the configuration given to an sharg::parser::add_option call is valid.
+    template <typename config_type>
+    void verify_option_config(config_type const & config)
```

Comment:
This should also work:

```suggestion
    void verify_option_config(config<auto> const & config)
```

Could save you the `template` line

If it doesn't work, I'd probably prefer

```suggestion
    template <typename t> // with a proper name
    void verify_option_config(config<t> const & config)
```